ontotext-yasgui-web-component/src/services/yasgui/configuration/yasgui-configuration-builder.ts

```diff
@@ -109,13 +109,13 @@ export class YasguiConfigurationBuilder {
 
   private initShortcuts(config: YasguiConfiguration): void {
     const keyboardShortcutDescriptions = KeyboardShortcutService.initKeyboardShortcutMapping();
-    const extraKeys = {};
-    keyboardShortcutDescriptions.forEach(keyboardShortcutDescription => {
-      keyboardShortcutDescription.keyboardShortcuts.forEach(keyboardShortcut => {
-        extraKeys[keyboardShortcut] = keyboardShortcutDescription.executeFunction;
-      });
-    });
-    config.yasguiConfig.yasqe.extraKeys = extraKeys;
+    config.yasguiConfig.yasqe.extraKeys = keyboardShortcutDescriptions.reduce((extra, keyboardShortcutDescription) => {
```

Comment:
Again, you can can avoid this nesting and complexity by flattening this by just mapping the keyboardShortcutDescriptions and then either reduce or loop-assign as in your first implementation.